actions-nightly-trigger/decide-trigger.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+# Decision logic for smart nightly trigger
+# Returns: 0 if should trigger, 1 if should skip
+# Outputs: writes trigger decision to result.txt
+
+set -e
+
+# Input parameters
+FORCE_TRIGGER="${FORCE_TRIGGER:-false}"
```

Comment:
What is the point of "force trigger"? This can be done directly on the actual "nightly" workflow, keeping things separated ("triggering" logic in cron/here / allowing manual triggering + workflow triggering as needed on the nightly itself)